drizzle-zod/tests/pg.test.ts

```diff
@@ -54,6 +56,12 @@ test('users insert invalid char', (t) => {
 	t.is(schema.safeParse({ ...testUser, initials: 'JoDo' }).success, false);
 });
 
+test('users insert invalid number type', (t) => {
+	const schema = createInsertSchema(users);
+
+	t.is(schema.safeParse({ ...testUser, numberOfDogs: 7.31 }).success, false);
+});
```

Comment:
this is the reason for the PR, today this test fails: parsing succeeds when it shouldn't, because postgres _will_ fail if you try to insert a float on an integer column